evm/rlp/transactions.py

```diff
@@ -36,22 +44,22 @@ class BaseTransaction(rlp.Serializable):
     ]
 
     @classmethod
-    def from_base_transaction(cls, transaction):
+    def from_base_transaction(cls, transaction: 'BaseTransaction') -> 'BaseTransaction':
         return rlp.decode(rlp.encode(transaction), sedes=cls)
 
     @property
-    def hash(self):
+    def hash(self) -> bytes:
         return keccak(rlp.encode(self))
 
     @property
-    def sender(self):
+    def sender(self) -> bytes:
```

Comment:
Similarly, we could create a custom `address` type (like outlined here in mypy docs http://mypy.readthedocs.io/en/latest/kinds_of_types.html#newtypes)

This would prevent us from accepting just *any* bytes value as an address.